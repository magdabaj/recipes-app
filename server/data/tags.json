[
  {
    "type": "dodatki",
    "values": ["zelatyna", "zakwas", "oliwki", "majonez", "ciasto francuskie"]
  },
  {
    "type": "alkohol",
    "values": ["advocaat", "brandy", "cydr", "rum", "wino"]
  },
  {
    "type": "herbata, kawa",
    "values": ["kawa", "herbata", "matcha", "nesquik", "horlicks"]
  },
  {
    "type": "przyprawy, ziola",
    "values": ["anyz", "bazylia", "cynamon", "imbir", "przyprawy korzenne", "mieta"]
  },
  {
    "type": "miody, syropy",
    "values": ["golden syrup", "melasa", "miod", "syrop klonowy"]
  },
  {
    "type": "bakalie",
    "values": [
      "mak",
      "migdaly",
      "orzechy",
      "rodzynki",
      "suszona zurawina",
      "suszone daktyle",
      "suszone figi",
      "suszone morele",
      "suszone sliwki",
      "suszone wisnie",
      "wiorki kokosowe"
    ]
  },
  {
    "type": "czekolada, znane slodycze",
    "values": [
      "amaretti",
      "chalwa",
      "czekolada",
      "ferrero rocher",
      "krakersy",
      "michalki",
      "milky way",
      "oreo",
      "rafaello",
      "reese",
      "snickers",
      "toffifee"
    ]
  },
  {
    "type": "masla, kremy, curdy",
    "values": ["maslo orzechowe", "nutella", "tahini"]
  },
  {
    "type": "warzywa, grzyby",
    "values": ["brokuly", "buraki", "cebula", "ciecierzyca", "czosnek", "marchewka", "pomidory"]
  },
  {
    "type": "owoce",
    "values": [
      "agrest",
      "ananasy",
      "maliny",
      "truskawki",
      "banany",
      "czeresnie",
      "mango",
      "pomarancze",
      "wisnie",
      "zurawina"
    ]
  },
  {
    "type": "smietany, jogurty, mleczka",
    "values": [
      "jogurt",
      "kwasna smietana",
      "mleko kokosowe",
      "mleko skondenowane",
      "mleko w proszku",
      "smietana kremowka"
    ]
  },
  {
    "type": "twarogi, serki, sery",
    "values": [
      "feta",
      "mascarpone",
      "mozarella",
      "parmezan",
      "ricotta",
      "ser zolty",
      "serek wiejski",
      "twarog"
    ]
  },
  {
    "type": "maki, platki, ziarna",
    "values": [
      "chia",
      "granola",
      "kakao",
      "maka grahama",
      "maka razowa",
      "maka orkiszowa",
      "maka zytnia",
      "muesli",
      "platki owsiane",
      "ryz",
      "sezam",
      "siemie lniane",
      "slonecznik"
    ]
  }
]
