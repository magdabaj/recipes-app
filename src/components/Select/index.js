/**
 *
 * Select
 *
 */

import React, { useState } from 'react';
import _ from 'lodash';
import '../../containers/RecipesFormContainer/index.css';
// import PropTypes from 'prop-types';
// import styled from 'styled-components';

function Select({ tags, tagType, onChange, error }) {
    const [tagValues, hideTagValues] = useState(false);
    const changeVisibility = () => hideTagValues(!tagValues);

    function sortTags(tagType) {
        return tags.map(tag => {
            if (tag.tagType === tagType) {
                return { tagValue: tag.tagValue, tagId: tag.id };
            }
        });
    }
    let sortedTags = _.compact(sortTags(tagType));

    return (
        <div className={'select'}>
            <label htmlFor={tagType} onClick={() => changeVisibility()} className={'select-tag-type'}>
                {tagType}
            </label>
            {tagValues
                ? sortedTags.map(tag => (
                    <div>
                        <input
                            type="radio"
                            id={tag.tagId}
                            name={'tagId'}
                            value={parseInt(tag.tagId)}
                            onChange={onChange}
                            className={'checkbox-input'}
                        />
                        <label htmlFor={tag.tagId} className={'checkbox-label'}>
                            {tag.tagValue}
                        </label>
                        <div className={'checkbox-control'} />
                    </div>
                ))
                : null}
        </div>
    );
}

Select.propTypes = {};

export default Select;
