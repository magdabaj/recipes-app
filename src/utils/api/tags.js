import "babel-polyfill"

const api = 'https://afternoon-plateau-23579.herokuapp.com/tags';
// const api = 'http://localhost:5000/tags'

export const fetchAllTags = async () => {
    const response = await fetch(api);
    const data = await response.json();
    if (response.status >= 400) {
        throw new Error(data.errors);
    }
    return data;
};