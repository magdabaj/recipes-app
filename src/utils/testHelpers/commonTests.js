const commonTests = (render, useFirstChildAsSnapshot = true) => {
    it('does not log errors in console', () => {
        const spy = jest.spyOn(global.console, 'error')
        render()
        expect(spy).not.toHaveBeenCalled()
    })

    it('should render and match the snapshot',  () => {
        const { container } = render()
        expect(container).not.toBeNull()
        if(useFirstChildAsSnapshot) expect(container.firstChild).not.toBeNull()
        expect(
            useFirstChildAsSnapshot ? container.firstChild : container,
        ).toMatchSnapshot()
    });
}

export default commonTests;