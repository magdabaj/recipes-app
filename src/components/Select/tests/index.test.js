import Select from "../index";
import { render } from "@testing-library/react";
import {fakeTags} from "../../../utils/testHelpers/fixtures/tags";
import React from "react";
import commonTests from "../../../utils/testHelpers/commonTests";

const renderSelect = () =>
    render(<Select tags={fakeTags} tagType={'alkohol'} onChange={jest.fn()}/>)

commonTests(renderSelect)