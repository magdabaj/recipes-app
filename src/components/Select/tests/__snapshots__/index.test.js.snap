// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match the snapshot 1`] = `
<div
  class="select"
>
  <label
    class="select-tag-type"
    data-testid="alkohol"
    for="alkohol"
  >
    alkohol
  </label>
</div>
`;
